File: TFM/control/graph/graph.m
data = csvread('../lectura/data.csv');

n = data(:,1);
t = data(:,2);
x_off = data(:,3);
x_zmp = data(:,4);
x_out = data(:,5);
x_angle = data(:,6);
xref = zeros(size(n));

% y_off = data(:,3);
% y_zmp = data(:,4);
% y_out = data(:,5);
% y_angle = data(:,6);
% yref = zeros(size(n));


%Sole borders
front = (120+55)/1000*ones(size(n));
back = -70/1000*ones(size(n));
left = (140+37.5)/1000*ones(size(n));
right = -(140+37.5)/1000*ones(size(n));

%% X REPRESENTATION
figure(1);
subplot(2,1,1)
title({'Experimento control T=0.03'; '[K Kp Ki Kd]=[18.89, 6.14, -3000, 10, 10]'});
hold on; box on
plot(t,xref,'k');
plot(t,x_off,'b');
plot(t,x_zmp,'g');
plot(t,x_out,'r');
plot(t,front,'k--',t,back,'k--');
legend('x_{ref}','x_{ref}-x_{zmp}','x_{zmp}','x_{out}');
ylabel('[meters]')

subplot(2,1,2)
plot(t,x_angle,'b');
legend('angle');
ylabel('Degrees');
xlabel('Time [s]');

%% Y REPRESENTATIONS
% figure(2); hold on;
% subplot(2,1,1)
% title({'Experimento control'; '[K Kp Ki Kd]=[18.89, 6.14, -3000, 10, 10]'});
% hold on
% plot(t,yref,'k');
% plot(t,y_off,'b');
% plot(t,y_zmp,'g');
% plot(t,y_out,'r');
% plot(t,left,'k--',n,right,'k--');
% legend('y_{ref}','y_{ref}-y_{zmp}','y_{zmp}','y_{out}');
% ylabel('[meters]')
% 
% subplot(2,1,2)
% hold on;
% plot(t,y_angle,'b');
% plot(t,-y_angle,'r');
% legend('angleRL','angleLL');
% ylabel('Degrees');

%% X-Y GRAPHS

% figure(3); hold on;
% plot(x_zmp,yzmp,'b');

File: TFM/control/graph/graph.py
import numpy as np
import matplotlib.pyplot as plt
import csv

data = np.loadtxt('/home/teo/data.csv', delimiter=",")
n = data[:,0]
t = data[:,1]
zmp = data[:,2]
y = data[:,3]
error = data[:,4]
ref = data[:,5]
u = data[:,6]
x1 = data[:,7]
x2 = data [:,8]
angle = data[:,9]

# Sole borders
front = ((120+55)/1000)*np.ones(n.size);
back = -70/1000*np.ones(n.size);
#left = (140+37.5)/1000*np.ones(size(n));
#right = -(140+37.5)/1000*np.ones(size(n));

fig = plt.figure()

plt.plot(t, zmp, 'g-', label= 'zmp')
plt.plot(t, y, 'r-', label = 'zmp_model')
plt.plot(t, error, 'b-', label = 'error')
plt.plot(t,ref,'k-', label = 'ref')
plt.plot(t,u,'m-', label = 'u')
plt.plot(t,x1,'y')
plt.plot(t,x2,'c')
plt.plot(t,angle,'y-', label = 'angle')
plt.plot(t,front,'k--',t,back,'k--');

plt.title('K=[12.55, 4.91], Ki=10, Kp=-1.5, Ku=-2.05')
plt.xlabel('t [s]')

plt.show()


